v5.7.0

iarna opened the pull request:
[CHANGELOG](https://github.com/npm/npm/blob/release-next/CHANGELOG.md)

zkat committed:
pkglock: automatically resolve conflicts

iarna committed:
auth, token: Make new commands support legacy auths

iarna committed:
token: Fix token revocation when otp is required

Previously, due to a bug, we didn't wait for token revocation to complete
before printing success. Further, if token revocation required an otp then
we didn't prompt for one.

Policy is that OTP _shouldn't_ be required for revocation, but allowing for
it should the registry require it is harmless.

zkat committed:
tar@4.1.1

iarna committed:
inflate-shrinkwrap: Support replacing resolved with file: urls

iarna committed:
actions: Correct logging about # of actions being run

iarna committed:
install: Only autoprune on install with lock file

iarna committed:
token: Fix CIDR list processing to handle comma separated items correctly

This got broken by having the config-type set to Array.  When it was a
String, it worked correctly.

iarna committed:
test: Fix lifecycle-path to work with externed npm-lifecycle

iarna committed:
fs-minipass@1.2.5

iarna committed:
profile: require password to change email

zkat committed:
ci: add new npm ci installer

iarna committed:
*: Switch from mkdirp to correctMkdir to preserve perms and owners

Switch to correctMkdir as much as possible to ensure that permissions on
directories are correct.

zkat committed:
ssri@5.1.0

This should fix the spurious EINTEGRITY issue

iarna committed:
maketest: Make test boilerplate use const/let instead of var

iarna committed:
prune: Fix bug where prune --production would remove dev deps from the lock file

zkat committed:
pkglock: fix integrity matching

zkat committed:
tar@4.3.3

zkat committed:
ssri@5.2.1

zkat committed:
cacache@10.0.2

zkat committed:
libcipm@1.3.2

isaacs committed:
auth: Add support for web-based logins

Reviewed-By: @iarna
Credit: @isaacs
PR-URL: https://github.com/npm/npm/pull/19780

kemitchell committed:
install: count contributors

PR-URL: https://github.com/npm/npm/pull/19817
Credit: @kemitchell
Reviewed-By: @iarna

zkat committed:
pacote@7.3.3

iarna committed:
install: Pull metadata filter out into its own module

iarna committed:
has-modern-meta: git deps do not need shasum or integrity

iarna committed:
install: Handle packages without install metadata gracefully

Currently the only way to get these is to use yarn.

What this means is that we:

Leave them untouched if you didn't as us to do a full install.

Upgrade their metadata if you DID ask us to do a full install.

While: Not messing with a lock file that only has version fields except for
modules we touched.

zkat committed:
ssri@5.2.4

zkat committed:
mississippi@2.0.0

zkat committed:
cacache@10.0.4

iarna committed:
diff-trees: Link-status is key in equivalency checking

iarna committed:
test: config-list: Make it work the same in lifecycles as out

iarna committed:
find-npm-prefix@1.0.2

Internal rewrite to not use bluebird

Credit: @iarna

iarna committed:
ini@1.3.5

Credit: @isaacs

iarna committed:
JSONStream@1.3.2

Credit: @dominictarr

iarna committed:
worker-farm@1.5.2

Credit: @rvagg

iarna committed:
readable-stream@2.3.4

Credit: @mcollina

iarna committed:
osenv@0.1.5

Credit: @isaacs

iarna committed:
npm-registry-couchapp@2.7.1

iarna committed:
marked@0.3.15

Credit: @joshbruce

iarna committed:
tap@11.1.1

Credit: @isaacs

iarna committed:
query-string@5.1.0

Credit: @sindresorhus

iarna committed:
semver@5.5.0

Credit: @isaacs

iarna committed:
uuid@3.2.1

Credit: @broofa

zkat committed:
libcipm@1.3.3

iarna committed:
doc: changelog for 5.7.0

iarna committed:
npm-profile@3.0.1

Add support for new web-based logins.

Credit: @isaacs

zkat commented on CHANGELOG.md:
```diff
@@ -1,3 +1,184 @@
+* [`17079c2a8`](https://github.com/npm/npm/commit/17079c2a880d3a6f8a67c8f17eedc7eb51b8f0f8)
+  Require password to change email through `npm profile`.
+  ([@iarna](https://github.com/iarna))
+* [`1d385ac0c`](https://github.com/npm/npm/commit/1d385ac0c35380b43fb5e453805f4850326902c0)
```
I would omit this one -- there was more stuff that was needed to get this working right, and `pacote@7.4.0` (the kinda-broken one) is the one that actually finishes adding all the stuff that needs to be there. Not to mention, I kinda removed most of the code from this commit as part of `npm-ci` integration. 🙈

zkat reviewed (approved):
One little comment that I'll leave up to you to decide whether to do anything about, but otherwise 👍

iarna committed:
install: Regenerate the logical tree after loading our lockfile

This fixes a problem where `removeObsoleteDep` wasn't able to walk the
logical tree when acting without a `node_modules` folder as none existed.

iarna committed:
install/deps: Pass our logger through to removeObsoleteDep from earliestInstallable

Historically earliestInstallable didn't have a logger, but since it does we
may as well pass it along.
